openmp/runtime/src/kmp_taskdeps.cpp

```diff
@@ -1030,6 +1030,12 @@ void __kmpc_omp_taskwait_deps_51(ident_t *loc_ref, kmp_int32 gtid,
                        __kmp_task_stealing_constraint);
   }
 
+  // Wait until the last __kmp_release_deps is finished before we free the
+  // current stack frame holding the "node" variable; once its nrefs count
+  // reaches 1, we're sure nobody else can try to reference it again.
+  while (node.dn.nrefs > 1)
```

Comment:
I'm not sure this is the right way to do that, especially using task yield. @jprotze might have a better idea.

Reply:
An alternative solution would be to just not use a stack-allocated node, but allocate it on the heap like everywhere else.  Then there'd still be a race which thread would be the one to free it, but that race would be harmless ...  Let me know if you'd like me to try this (or any other) approach.